pkg/deploy/generator/config_generator_test.go

```diff
@@ -175,23 +138,24 @@ func TestGenerate_fromConfigWithUpdatedImageRef(t *testing.T) {
 		t.Fatalf("Expected non-nil config")
 	}
 
-	if config.LatestVersion != 2 {
-		t.Fatalf("Expected config LatestVersion=2, got %d", config.LatestVersion)
+	if config.Status.LatestVersion != 2 {
+		t.Fatalf("Expected config LatestVersion=2, got %d", config.Status.LatestVersion)
 	}
 
-	if e, a := newRepoName, config.Template.ControllerTemplate.Template.Spec.Containers[0].Image; e != a {
-		t.Fatalf("Expected container image %s, got %s", e, a)
+	if got, exp := newRepoName, config.Spec.Template.Spec.Containers[0].Image; got != exp {
```

Comment:
e=expected, a=actual.  I prefer `e` and `a`, but if you're going to change it make sure you keep it correct.  I think you've accidentally swapped them.